frontend/src/components/WildcardSelect.vue

```diff
@@ -0,0 +1,221 @@
+      item-color="cyan darken-2"
+      :label="wildcardSelectLabel"
+      :items="wildcardSelectItemObjects"
+      return-object
+      v-model="wildcardSelectedValue"
+      :error-messages="getErrorMessages('wildcardSelectedValue')"
+      @input="onInput"
+      @blur="$v.wildcardSelectedValue.$touch()"
+      :hint="wildcardSelectHint"
+      persistent-hint
+    >
+      <template v-slot:item="{ item }">
+        <v-list-item-content>
+          <v-list-item-title>
+            <span v-if="item.startsWithWildcard">&lt;prefix&gt;</span>
+            <span>{{item.text}}</span>
+            <span v-if="item.endsWithWildcard">&lt;suffix&gt;</span>
+          </v-list-item-title>
+        </v-list-item-content>
+      </template>
+    </v-select>
+    <v-text-field
+      v-if="wildcardSelectedValue.isWildcard"
+      :class="textFieldClass"
+      color="cyan darken-2"
+      :label="wildcardTextFieldLabel"
+      v-model="wildcardVariablePart"
+      :error-messages="getErrorMessages('wildcardVariablePart')"
+      @input="onInput"
+      @blur="$v.wildcardVariablePart.$touch()"
+      ></v-text-field>
+  </div>
+</template>
+
+<script>
+import startsWith from 'lodash/startsWith'
+import endsWith from 'lodash/endsWith'
+import map from 'lodash/map'
+import trim from 'lodash/trim'
+import filter from 'lodash/filter'
+import head from 'lodash/head'
+import { getValidationErrors } from '@/utils'
+import { required, requiredIf } from 'vuelidate/lib/validators'
+
+const validations = {
+  wildcardVariablePart: {
+    required: requiredIf(function () {
+      return this.wildcardSelectedValue.isWildcard
+    })
+  },
+  wildcardSelectedValue: {
+    required
+  }
+}
+
+export default {
+  name: 'wildcard-select',
+  props: {
+    wildcardSelectItems: {
+      type: Array
+    },
+    wildcardSelectLabel: {
+      type: String
+    },
+    value: {
+      type: String
+    }
+  },
+  data () {
+    return {
+      wildcardVariablePart: undefined,
+      wildcardSelectedValue: {},
+      valid: undefined
+    }
+  },
+  computed: {
+    validationErrors () {
+      return {
+        wildcardVariablePart: {
+          required: `${this.wildcardTextFieldLabel} is required`
+        },
+        wildcardSelectedValue: {
+          required: `${this.wildcardSelectLabel} is required`
+        }
+      }
+    },
+    textFieldClass () {
+      return this.wildcardSelectedValue.startsWithWildcard ? 'textFieldStartClass' : 'textFieldEndClass'
+    },
+    wildcardSelectItemObjects () {
+      return map(this.wildcardSelectItems, item => {
+        let startsWithWildcard = false
+        let endsWithWildcard = false
+
+        let selectedValue = item
+        if (startsWith(item, '*')) {
+          startsWithWildcard = true
+        }
+        if (endsWith(item, '*')) {
+          endsWithWildcard = true
+        }
+        selectedValue = trim(item, '*')
+
+        return {
+          text: selectedValue,
+          value: selectedValue,
```

Comment:
why do we need both text and value with the same values?

Reply:
leftover from before we had custom slots. v-select expects text / value in case of item objects. will remove it, we can re-use value attribute in the slot

Reply:
ok I eneded up leaving it as is with value and text, as we require text for the select compinent and I don't want to treat `text` as value in my code.